src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.GenerateImplicitNamespaceImports.targets

```diff
@@ -0,0 +1,66 @@
+***********************************************************************************************
+-->
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!--
+    ============================================================
+                                     GenerateImplicitNamespaceImports
+
+    Generates implicit namespace imports source to intermediate directory for C# projects
+    ============================================================
+    -->
+  <PropertyGroup>
+    <GeneratedImplicitNamespaceImportFile Condition="'$(GeneratedImplicitNamespaceImportFile)' ==''">$(IntermediateOutputPath)$(MSBuildProjectName).ImplicitNamespaceImports$(DefaultLanguageSourceExtension)</GeneratedImplicitNamespaceImportFile>
+    <DisableImplicitNamespaceImports_DotNet Condition="'$(DisableImplicitNamespaceImports_DotNet)' == '' and '$(DisableImplicitNamespaceImports)' != 'true'">false</DisableImplicitNamespaceImports_DotNet>
+  </PropertyGroup>
+
+  <ItemGroup Condition=" '$(DisableImplicitNamespaceImports_DotNet)' != 'true'
+                      and '$(TargetFrameworkIdentifier)' == '.NETCoreApp'
+                      and $([MSBuild]::VersionGreaterThanOrEquals($(_TargetFrameworkVersionWithoutV), '6.0'))">
+    <Import Include="System" />
+    <Import Include="System.Collections.Generic" />
+    <Import Include="System.IO" />
+    <Import Include="System.Linq" />
+    <Import Include="System.Net.Http" />
+    <Import Include="System.Threading.Tasks" />
+  </ItemGroup>
+
+  <!--
+    Note that this must run before every invocation of CoreCompile to ensure that all compiler
+    runs see the generated assembly info. There is at least one scenario involving Xaml
+    where CoreCompile is invoked without other potential hooks such as Compile or CoreBuild,
+    etc., so we hook directly on to CoreCompile. Furthermore, we  must run *after*
+    PrepareForBuild to ensure that the intermediate directory has been created.
+
+    Targets that generate Compile items are also expected to run before
+    BeforeCompile targets (common targets convention).
+   -->
+  <Target Name="GenerateImplicitNamespaceImports"
+          BeforeTargets="BeforeCompile;CoreCompile"
+          DependsOnTargets="PrepareForBuild;CoreGenerateImplicitNamespaceImports"
+          Condition="'$(DisableImplicitNamespaceImports)' != 'true'" />
+
+  <Target Name="CoreGenerateImplicitNamespaceImports"
+          Outputs="$(GeneratedImplicitNamespaceImportFile)">
+    <ItemGroup>
+      <_ImportFileLine Include="// %3Cautogenerated />"/>
+      <_ImportFileLine Include="global using global::%(Import.Identity)%3B"/>
+    </ItemGroup>
+
+    <WriteLinesToFile Lines="@(_ImportFileLine)" File="$(GeneratedImplicitNamespaceImportFile)" Overwrite="true" WriteOnlyWhenDifferent="true" />
+
+    <ItemGroup>
+      <Compile Include="$(GeneratedImplicitNamespaceImportFile)" />
+      <FileWrites Include="$(GeneratedImplicitNamespaceImportFile)" />
+    </ItemGroup>
+  </Target>
```

Comment:
I applied this locally, and something is not quite right here - it applies to non net6.0 builds as well, e.g. I tried building a netcoreapp3.1
![image](https://user-images.githubusercontent.com/4403806/124074688-359f3b00-da87-11eb-9dc4-f1dcb2e85d82.png)

Reply:
Ah I see, I should add a condition to only write when the itemgroup is not empty.

Reply:
The logic is updated to handle when the itemgroup is empty. A test case was also added.